src/FileType_XDMF.f90

```diff
@@ -0,0 +1,1324 @@
+! Public members
+PUBLIC :: XDMFFileType
+PUBLIC :: XDMFMeshType
+PUBLIC :: XDMFTopologyList
+PUBLIC :: ASSIGNMENT(=)
+
+!> The module name
+CHARACTER(LEN=*),PARAMETER :: modName='FILETYPE_XDMF'
+
+!> Exception handler for the module
+TYPE(ExceptionHandlerType),SAVE :: eXDMF
+
+!> Parameter list that holds XDMF topology names, ids, etc.
+TYPE(ParamType),SAVE :: XDMFTopologyList
+
+!> Type to hold the vertices that make up a mesh cell
+TYPE :: XDMFCell
+  !> The cell type id followed by the vertex ids
+  !> XDMF ID, v1, v2, ..., v_n
+  INTEGER(SLK), ALLOCATABLE :: vertex_list(:)
+ENDTYPE XDMFCell
+
+!> Type to hold a list of cell IDs that make up a named set.
+TYPE :: XDMFCellSet
+  !> The name of the set
+  TYPE(StringType) :: name
+  !> The cell IDs
+  INTEGER(SLK), ALLOCATABLE :: cell_list(:)
+ENDTYPE XDMFCellSet
+
+!> Type to hold the XDMF mesh data
+TYPE :: XDMFMeshType
+  !> The name of the mesh
+  TYPE(StringType) :: name
+  !> If the mesh cells are all the same topology
+  LOGICAL(SBK) :: singleTopology=.FALSE.
+  !> The bounding box for the mesh
+  !> xmin, xmax, ymin, ymax
+  REAL(SDK) :: boundingBox(4) = 0.0_SDK
+  !> The vertices that compose the mesh
+  !> Looks like:
+  !> x1, x2, x3, ..., xn
+  !> y1, y2, y3, ..., yn
+  !> z1, z2, z3, ..., zn
+  !> Therefore vertices will be of shape (3, N)
+  REAL(SDK), ALLOCATABLE :: vertices(:, :)
+  !> The mesh cells
+  TYPE(XDMFCell), ALLOCATABLE :: cells(:)
+  !> Material for each mesh cell
+  INTEGER(SNK), ALLOCATABLE :: material_ids(:)
+  !> Named sets within the mesh
+  TYPE(XDMFCellSet), ALLOCATABLE :: cell_sets(:)
+  !> Child and parent meshes
+  TYPE(XDMFMeshType), POINTER :: parent => NULL(), children(:) => NULL()
+  CONTAINS
+    !> @copybrief XDMFMeshType::clear_XDMFMeshType
+    !> @copydoc XDMFMeshType::clear_XDMFMeshType
+    PROCEDURE,PASS :: clear => clear_XDMFMeshType
+    !> @copybrief XDMFMeshType::distanceToLeaf_XDMFMeshType
+    !> @copydoc XDMFMeshType::distanceToLeaf_XDMFMeshType
+    PROCEDURE,PASS :: distanceToLeaf => distanceToLeaf_XDMFMeshType
+    !> @copybrief XDMFMeshType::recomputeBoundingBox_XDMFMeshType
+    !> @copydoc XDMFMeshType::recomputeBoundingBox_XDMFMeshType
+    PROCEDURE,PASS :: recomputeBoundingBox => recomputeBoundingBox_XDMFMeshType
+ENDTYPE XDMFMeshType
+
+!> The XDMF File type
+TYPE,EXTENDS(BaseFileType) :: XDMFFileType
+!
+!List of type bound procedures
+!
+!Import and export are the only procedures used.
+!open, close, and delete are members of the base type.
+  CONTAINS
+    !> @copybrief FileType_XDMF::fopen_XDMFFileType
+    !> @copydoc FileType_XDMF::fopen_XDMFFileType
+    PROCEDURE,PASS :: fopen => fopen_XDMFFileType
+    !> @copybrief FileType_XDMF::fclose_XDMFFileType
+    !> @copydoc FileType_XDMF::fclose_XDMFFileType
+    PROCEDURE,PASS :: fclose => fclose_XDMFFileType
+    !> @copybrief FileType_XDMF::fdelete_XDMFFileType
+    !> @copydoc FileType_XDMF::fdelete_XDMFFileType
+    PROCEDURE,PASS :: fdelete => fdelete_XDMFFileType
```

Comment:
Is any information used from the BaseFileType? If not, I don't see the XDMF should be derived from it. Right now, you have to implement these deferred procedures (and they don't do anything). Maybe XDMF is a data structure or format that is not suitable to be defined as a file type?